% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/fragCounter.R
\name{bam.cov.skel}
\alias{bam.cov.skel}
\title{Get coverage as GRanges from BAM using exons as tiles}
\usage{
bam.cov.skel(
  bam.file,
  skeleton,
  chunksize = 1e+05,
  min.mapq = 20,
  verbose = TRUE,
  reference = NULL,
  max.tlen = 10000,
  st.flag = fragCounter::ST.FLAG,
  fragments = TRUE,
  do.gc = FALSE,
  use.skel = FALSE
)
}
\arguments{
\item{bam.file}{string Input BAM file}

\item{skeleton}{string Input data.table with intervals for which there is coverage data}

\item{chunksize}{integer Size of window (default = 1e5)}

\item{min.mapq}{integer Minimim map quality reads to consider for counts (default = 20)}

\item{verbose}{boolean Flag to increase vebosity (default = TRUE)}

\item{max.tlen}{integer Maximum paired-read insert size to consider (default = 1e4)}

\item{st.flag}{string Samtools flag to filter reads on (default = '-f 0x02 -F 0x10')}

\item{fragments}{boolean flag (default = FALSE) detremining whether to compute fragment (i.e. proper pair footprint aka insert) density or read density}

\item{do.gc}{boolean Flag to execute garbage collection via 'gc()' (default = FALSE)}

\item{use.skel}{boolean flag If false then default exome skeleton from gencode is used, if TRUE, user defined skeleton is used}
}
\value{
GRanges of user defined tiles across seqlengths of bam.file with meta data field $counts specifying fragment counts centered (default = TRUE)
in the given bin.
}
\description{
Quick way to get tiled coverage via piping to samtools (e.g. ~10 CPU-hours for 100bp tiles, 5e8 read pairs)

Gets coverage for user-defined regions of the genome, pulling "chunksize" records at a time and incrementing bin
}
\author{
Trent Walradt
}
